### PR TITLE
etcdserver: etcd should fall back to proxy again if proxy data is detected

### LINKED ISSUE #2504: etcdserver: create etcd member directory when fallback to proxy
etcd output:

```
etcd: listening for client requests on https://0.0.0.0:2379
datadir is valid for the 2.0.1 format
etcd: stopping listening for client requests on https://0.0.0.0:2379
etcd: stopping listening for peers on https://0.0.0.0:2380
etcd: cannot write to member directory: open /root/etcd/member/.touch: no such file or directory

### PR DESCRIPTION
**Pull request by xiang90**
Fix #2504 

Tested manually.

### CODE REVIEW

File: etcdserver/server.go [hunk 1]
```diff
@@ -147,9 +147,14 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 	if err != nil {
 		return nil, err
 	}
-	if walVersion == wal.WALUnknown {
+	switch walVersion {
+	case wal.WALUnknown:
 		return nil, fmt.Errorf("unknown wal version in data dir %s", cfg.DataDir)
+	case wal.WALv2_0Proxy:
+		// skip the discovery path and return the previous full cluster error again.
+		return nil, discovery.ErrFullCluster
```
Comment (yichengq) [inline etcdserver/server.go]: I think it is better to use a separate error. It is unclear about why wal.WALv2_0Proxy leads to discovery.ErrFullCluster. Considering user who bootstrap etcd member from the same data dir which is used for proxy before, the error `discovery.ErrFullCluster` is weird.

Comment (xiang90) [inline etcdserver/server.go]: i do not think it is wired. we simply by pass the checking again and return the correct error it should return.

**Comment by xiang90** [comment]
@yichengq PTAL.

File: etcdmain/etcd.go [hunk 2]
```diff
@@ -53,7 +62,17 @@ func Main() {
...
 
-	shouldProxy := cfg.isProxy()
+	if cfg.dir == "" {
+		cfg.dir = fmt.Sprintf("%v.etcd", cfg.name)
+		log.Printf("etcd: no data-dir provided, using default data-dir ./%s", cfg.dir)
+	}
+
+	which := identifyDataDirOrDie(cfg.dir)
+	if which != dirEmpty {
+		log.Printf("etcd: %v dir already exists, starting etcd %v...", which, which)
```
Comment (yichengq) [inline etcdmain/etcd.go]: log.Printf("etcd: dir %s/%v already exists, starting etcd %v...", cfg.dir, which, which)
user can have idea which dir etcd is talking about

Comment (xiang90) [inline etcdmain/etcd.go]: x dir is a type not a path. `member dir` exists -> member; `proxy dir` exists -> proxy.

Comment (xiang90) [inline etcdmain/etcd.go]: reworked this message a little bit anyway.

**Comment by yichengq** [comment]
lgtm if intergration test pass.

**Comment by xiang90** [comment]
@yichengq I guess it will never pass. help me out travis...

**Pull request merged**